<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget name="Main Window" parent="main_window">
  <Tab tab_name="tab1" containers="1">
   <Container>
    <DockSplitter orientation="-" sizes="0.324247;0.675753" count="2">
     <DockSplitter orientation="|" sizes="0.500409;0.499591" count="2">
      <DockArea name="...">
       <plot flip_x="false" flip_y="false" mode="TimeSeries" style="Lines">
        <range right="6833.201454" top="32.480232" left="6733.300799" bottom="-104.876598"/>
        <limitY/>
        <curve name="/peacetolero/force_approach/force_error/data[0]" color="#1f77b4"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_x="false" flip_y="false" mode="TimeSeries" style="Lines">
        <range right="6833.201454" top="0.026250" left="6733.300799" bottom="-0.026250"/>
        <limitY/>
        <curve name="/peacetolero/force_approach/force_error/data[1]" color="#d62728"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter orientation="|" sizes="0.500409;0.499591" count="2">
      <DockSplitter orientation="-" sizes="0.5;0.5" count="2">
       <DockArea name="...">
        <plot flip_x="false" flip_y="false" mode="TimeSeries" style="Lines">
         <range right="6833.206484" top="0.292916" left="6733.305362" bottom="-0.029836"/>
         <limitY/>
         <curve name="/peacetolero/alpha_force_sensor/wrench/force/z" color="#1ac938"/>
         <curve name="/peacetolero/alpha_force_sensor/wrench/force/x" color="#f14cc1"/>
         <curve name="/peacetolero/alpha_force_sensor/wrench/force/y" color="#ff7f0e"/>
         <curve name="/peacetolero/cmd_vel/linear/x" color="#1f77b4"/>
         <curve name="/peacetolero/cmd_vel/angular/z" color="#d62728"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot flip_x="false" flip_y="false" mode="TimeSeries" style="Lines">
         <range right="6841.020327" top="54.217859" left="6741.022479" bottom="47.929901"/>
         <limitY/>
         <curve name="/joint_states/peacetolero/wheel_back_left_joint/position" color="#1ac938"/>
         <curve name="/joint_states/peacetolero/wheel_front_left_joint/position" color="#ff7f0e"/>
         <curve name="/joint_states/peacetolero/wheel_front_right_joint/position" color="#f14cc1"/>
         <curve name="/joint_states/peacetolero/wheel_back_right_joint/position" color="#9467bd"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockArea name="...">
       <plot flip_x="false" flip_y="false" mode="TimeSeries" style="Lines">
        <range right="6833.201454" top="0.032122" left="6733.300799" bottom="-0.084051"/>
        <limitY/>
        <curve name="/peacetolero/force_approach/force_error/data[5]" color="#9467bd"/>
        <curve name="/peacetolero/force_approach/force_error/data[6]" color="#17becf"/>
        <curve name="/peacetolero/force_approach/force_error/data[7]" color="#bcbd22"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <parameters delimiter="0" time_axis=""/>
  </plugin>
  <plugin ID="DataLoad MCAP"/>
  <plugin ID="DataLoad ROS2 bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
   <selected_topics value=""/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS2 Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
   <selected_topics value="/peacetolero/alpha_force_sensor;/peacetolero/cmd_vel;/peacetolero/force_approach/force_error;/joint_states"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;--]]&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. --]]&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value &#xa;  &#xa;--]]&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS2 Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS2 Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

